pkg/sqlmigrations/migrations.go

```diff
@@ -600,221 +579,37 @@ func populateVersionSetting(ctx context.Context, r runner) error {
 }
 
 func addRootUser(ctx context.Context, r runner) error {
-	// System tables can only be modified by a privileged internal user.
-	session := r.newRootSession(ctx)
-	defer session.Finish(r.sqlExecutor)
-
 	// Upsert the root user into the table. We intentionally override any existing entry.
-	const upsertRootStmt = `UPSERT INTO system.users (username, "hashedPassword") VALUES ($1, '')`
+	const upsertRootStmt = `
+	        UPSERT INTO system.users (username, "hashedPassword", "isRole") VALUES ($1, '', false)
+	        `
 
 	pl := tree.MakePlaceholderInfo()
 	pl.SetValue("1", tree.NewDString(security.RootUser))
-	res, err := r.sqlExecutor.ExecuteStatementsBuffered(session, upsertRootStmt, &pl, 1)
-	if err == nil {
-		res.Close(ctx)
-	}
-	return err
+	return runStmtAsRootWithRetry(ctx, r, upsertRootStmt, &pl)
 }
 
-func addDefaultMetaAndLivenessZoneConfigs(ctx context.Context, r runner) error {
-	defaultTTLSeconds := config.DefaultZoneConfig().GC.TTLSeconds
-
-	// Retrieve the existing .meta zone config.
-	metaZone, err := getZoneConfig(ctx, r, "RANGE meta")
-	if err != nil {
-		return err
-	}
-	// Update the GC TTL seconds if it still at the default setting.
-	if metaZone.GC.TTLSeconds == defaultTTLSeconds {
-		metaZone.GC.TTLSeconds = 60 * 60 // 1h
-	}
-	if err := upsertZoneConfig(ctx, r, keys.MetaRangesID, metaZone); err != nil {
-		return err
-	}
-
-	// The liveness range was previously covered by the ".system" zone. Grab the
-	// existing ".system" zone (if any) for modification.
-	livenessZone, err := getZoneConfig(ctx, r, "RANGE system")
-	if err != nil {
-		return err
-	}
-	// We set the .liveness zone config regardless, but only update the TTL
-	// seconds if it is still at the default setting.
-	if livenessZone.GC.TTLSeconds == defaultTTLSeconds {
-		livenessZone.GC.TTLSeconds = 10 * 60 // 10m
-	}
-	return upsertZoneConfig(ctx, r, keys.LivenessRangesID, livenessZone)
-}
-
-func upsertZoneConfig(ctx context.Context, r runner, id uint32, zone config.ZoneConfig) error {
```

Comment:
Zone config manipulation is no longer used in migrations, but it's likely to be used again. Rather than delete all this, I could move it to a "helpers" file for future use.